editoast/src/core/v2/simulation.rs

```diff
@@ -0,0 +1,145 @@
+use std::collections::BTreeMap;
+
+use chrono::DateTime;
+use chrono::Utc;
+use editoast_schemas::rolling_stock::EffortCurves;
+use editoast_schemas::rolling_stock::Gamma;
+use editoast_schemas::rolling_stock::RollingResistance;
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::core::{AsCoreRequest, Json};
+use crate::schema::utils::Identifier;
+use crate::schema::v2::trainschedule::MarginValue;
+use crate::schema::v2::trainschedule::TrainScheduleOptions;
+use crate::schema::v2::trainschedule::{Comfort, Distribution};
+use crate::views::v2::path::TrackRange;
+use crate::views::v2::train_schedule::CompleteReportTrain;
+use crate::views::v2::train_schedule::Mrsp;
+use crate::views::v2::train_schedule::ReportTrain;
+use crate::RollingStockModel;
+use derivative::Derivative;
+use std::hash::Hash;
+
+#[derive(Debug, Serialize, Derivative)]
+#[derivative(Hash)]
+pub struct PhysicRollingStock {
+    pub effort_curves: EffortCurves,
+    pub base_power_class: Option<String>,
+    /// Length of the rolling stock in mm
+    pub length: u64,
+    /// Maximum speed of the rolling stock in m/s
+    #[derivative(Hash(hash_with = "editoast_common::hash_float::<3,_>"))]
+    pub max_speed: f64,
+    // Time in ms
+    pub startup_time: u64,
+    #[derivative(Hash(hash_with = "editoast_common::hash_float::<5,_>"))]
+    pub startup_acceleration: f64,
+    #[derivative(Hash(hash_with = "editoast_common::hash_float::<5,_>"))]
+    pub comfort_acceleration: f64,
+    pub gamma: Gamma,
+    #[derivative(Hash(hash_with = "editoast_common::hash_float::<5,_>"))]
+    pub inertia_coefficient: f64,
+    /// Mass of the rolling stock in kg
+    pub mass: u64,
+    pub rolling_resistance: RollingResistance,
+    /// Mapping of power restriction code to power class
+    #[serde(default)]
+    pub power_restrictions: BTreeMap<String, String>,
+    /// The time the train takes before actually using electrical power (in miliseconds). Is null if the train is not electric.
+    pub electrical_power_startup_time: Option<u64>,
+    /// The time it takes to raise this train's pantograph in seconds. Is null if the train is not electric.
+    /// The time is in miliseconds.
+    pub raise_pantograph_time: Option<u64>,
```

Comment:
How about grouping these 2 properties into one `struct` (maybe with `#[serde(flatten)]`)? That would mean only one `Option` to handle in the code for when it's electric.

Reply:
One field can be null while the other is not, we change the doc.